traitsui/tests/editors/test_image_enum_editor.py

```diff
@@ -0,0 +1,488 @@
+            editor = self.setup_ui(model, formatted_view)
+
+            self.assertEqual(editor.names, ["TOP LEFT", "TOP RIGHT"])
+            self.assertEqual(
+                editor.mapping,
+                {"TOP LEFT": "top left", "TOP RIGHT": "top right"}
+                )
+            self.assertEqual(
+                editor.inverse_mapping,
+                {"top left": "TOP LEFT", "top right": "TOP RIGHT"}
+            )
+
+            model.possible_values = ["top right", "top left"]
+
+            self.assertEqual(editor.names, ["TOP RIGHT", "TOP LEFT"])
+            self.assertEqual(
+                editor.mapping,
+                {"TOP RIGHT": "top right", "TOP LEFT": "top left"}
+            )
+            self.assertEqual(
+                editor.inverse_mapping,
+                {"top right": "TOP RIGHT", "top left": "TOP LEFT"}
+            )
+
+    @unittest.skip("Issue enthought/traitsui#844")
+    def test_simple_editor_mapping_values(self):
+        self.check_enum_mappings_value_change("simple")
+
+    @unittest.skip("Issue enthought/traitsui#844")
+    def test_simple_editor_mapping_name(self):
+        self.check_enum_mappings_name_change("simple")
+
+    def test_custom_editor_mapping_values(self):
+        self.check_enum_mappings_value_change("custom")
+
+    def test_custom_editor_mapping_name(self):
+        self.check_enum_mappings_name_change("custom")
+
+    def test_readonly_editor_mapping_values(self):
+        self.check_enum_mappings_value_change("readonly")
+
+    def test_readonly_editor_name(self):
+        class PossibleEnumModel(HasTraits):
+            value = value = Enum('top left', 'top right')
+            possible_values = List(['top left', 'top right'])
+
+        formatted_view = View(
+            UItem(
+                'value',
+                editor=ImageEnumEditor(
+                    name="object.possible_values",
+                    format_func=lambda v: v.upper(),
+                    prefix='@icons:',
+                    suffix='_origin',
+                    path='dummy_path',
+                ),
+                style="readonly",
+            )
+        )
+        model = PossibleEnumModel()
+
+        with store_exceptions_on_all_threads():
+            editor = self.setup_ui(model, formatted_view)
+
+            # Readonly editor doesn't set up full mapping, only check that
+            # str_value is mapped as expected
+            self.assertEqual(model.value, "top left")
+            self.assertEqual(editor.str_value, "TOP LEFT")
+
+
+@skip_if_null
+class TestSimpleImageEnumEditor(unittest.TestCase):
+
+    def setup_gui(self, model, view):
+        gui = GUI()
+        ui = model.edit_traits(view=view)
+        self.addCleanup(ui.dispose)
+
+        gui.process_events()
+        editor = ui.get_editors("value")[0]
+        control = editor.control
+
+        return gui, control
+
+    @unittest.skipIf(is_linux, "Issue enthought/traitsui#854")
```

Comment:
#854 only applies to Qt, or did I misunderstand? So this should `skipIf(is_linux and is_backend_qt4)`?